lightbend-telemetry/lightbend-telemetry-java-mvn/README.md

```diff
@@ -0,0 +1,100 @@
+# Integrating Lagom with Lightbend Telemetry (Cinnamon)
+
+[Telemetry](https://developer.lightbend.com/docs/cinnamon/current/home.html) (also known as "Cinnamon"), part of Lightbend Enterprise Suite’s Intelligent Monitoring feature set, is a suite of insight tools that provides a view into the workings of our distributed platforms. This view allows developers and operations to respond quickly to problems, track down unexpected behavior and even tune your system. As a result, you can deploy your applications to production with confidence.
+
+Lightbend Telemetry [integrates with Lagom](https://developer.lightbend.com/docs/cinnamon/current/instrumentations/lagom/lagom.html) to provide circuit breaker metrics, HTTP client and server metrics, and request tracing.
+
+This example adapts the [instructions for integrating Telemetry into a Lagom Java service with sbt](https://developer.lightbend.com/docs/cinnamon/current/getting-started/lagom_java.html), showing how to do the same using Maven.
+
+## Prerequisites
+
+The following must be installed for these instructions to work:
+
+* Java 8
+* Maven 3.3 or higher
+* Bintray credentials
+
+### Bintray credentials
+
+Follow [these instructions](https://www.lightbend.com/product/lightbend-reactive-platform/credentials) to set up your Bintray credentials for Maven.
+
+## Testing the recipe
+
+Lagom has a special development mode for rapid development, and does not fork the JVM when using the `lagom:runAll` or `lagom:run` commands in Maven. A forked JVM is necessary to gain metrics for actors and HTTP calls, since those are provided by the Cinnamon Java Agent. This example uses the [`exec-maven-plugin`](https://www.mojohaus.org/exec-maven-plugin/) to run the Lagom production server, enabling the complete set of metrics to be displayed.
```

Comment:
This is a bit `meh`.

We're saying this recipe demonstrates using cinnamon in lagom but only for a single service and out of dev mode.

@pvlugter @bantonsson would it be possible to add the cinnamon agent on the JVM running `sbt` or `mvn` using `.jvmopts` or `MVN_OPTS`?

Reply:
I'll merge. We can continue the discussion somewhere else.

Reply:
Yes, should certainly be improved. And we've been here before with agents and Play dev run.

So the current plan is to have a classloader hook in the dev run, so Cinnamon agent can be added as a classloader transformation instead. This is how we used to have the Typesafe Console integration with Play dev run. This setting was removed. Issue to bring it back is here:

https://github.com/playframework/playframework/issues/8206

For Activator, which had both the Typesafe Console instrumentation (when it was open sourced) and integration with New Relic and AppDynamics agents, there was a forked dev run:

https://github.com/playframework/playframework/pull/3835